ccm_web/server/src/api/api.utils.ts

```diff
@@ -0,0 +1,18 @@
+import { HTTPError } from 'got'
+import { APIErrorData } from './api.interfaces'
+
+import baseLogger from '../logger'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+export function handleAPIError (error: unknown): APIErrorData {
+  if (error instanceof HTTPError) {
+    const { statusCode, statusMessage } = error.response
+    logger.error(`Received unusual status code ${String(statusCode)}`)
+    logger.error(`Response body: ${JSON.stringify(statusMessage)}`)
+    return { statusCode, message: `Error(s) from Canvas:  ${statusMessage as string}` }
```

Comment:
Okay, so here's where I'm at on the `body` question currently.

When there are errors reported here, these will be errors that have fallen through the cracks, that we may want to handle explicitly in future versions of the code. Because of that, I don't know if it's necessary to provide this information to the client. Errors that client will need to know lots of detail about should be caught by the validation code (see the `dto`). Instead, we can log the body so that someone responding to production incidents can find and use in creating an issue.

Thus, so long as we do log the body somewhere, I am okay with deleting `CanvasService.parseErrorBody` (and the related type guards) and relying on the `statusMessage` as what we deliver to the client. Doing this also has an added benefit of us not having to handle all possible shapes of Canvas error responses (of which I've seen a few, and which could conceivably change).

I'm willing to discuss further as needed, @pushyamig, thanks!